Feature/#182 refactor gamemanager

leroykorterink opened the pull request.

Commit:
Refactor GameManager for easier lobby setup #182

Commit:
Add rendereable colliders to all towers #186

Commit:
Add scene updates and rename material #186

leroykorterink committed:
Merge branch 'develop' into feature/#182-refactor-gamemanager

hassanAlAwwadi commented on src/Assets/LightSkyDefense/Scripts/GameStates/GameState.cs:
```diff
@@ -3,10 +3,33 @@
     private void Awake()
     {
-        _gameManager = GetComponent<GameManager>();
+        GameManager = GetComponent<GameManager>();
```
```suggestion
        GameManager = GameManager.Instance;
```

Commit:
Fix fallback controller #182

Commit:
No need to get GameManager in GameState #182

Commit:
Update scene and fix SpawnEnemy asset #186

Commit:
Always destroy preview in fallback #182

Commit:
Use coroutine to check for enemies #182

Commit:
Add infinity state as initial state #182

Commit:
Merge remote-tracking branch 'origin/develop' into feature/#182-refactor-gamemanager

Commit:
Update scene #182

hassanAlAwwadi committed:
Fix minor issues during VR testing #182

Commit:
Move files out of resources directory #182

Commit:
Merge remote-tracking branch 'origin/develop' into feature/#182-refactor-gamemanager

Commit:
Move new enemy prefabs out of resources #182

Commit:
Remove unnecessary method #182

Commit:
Move files and update enemy steering behaviour #182 #179

Enemy now uses a combination of flock, wander, and path following to steer. This looks a more realistic and works using only the Rigidbody. Any forces that are applied to the rigidbody will work just fine. The "PathFollower" game object is now integrated in the PathFollow steering behaviour to reduce the amount of objects in the scene.

Commit:
Update default max neighbours in flocking #182

Commit:
Remove unused script #182

Commit:
Update type for layer mask #182

Commit:
Remove unused variables #182

Maxboef commented on src/Assets/LightSkyDefense/Scripts/Common/Steering/Behaviours/FollowPath.cs:
```diff
@@ -0,0 +1,97 @@
+    private int _currentWaypointIndex;
+
+    /// <summary>
+    /// Rigidbody is used to progress along the path using the velocity's magnitude
```
Summary?

Commit:
Add enemy reach end of path #182

Maxboef commented on src/Assets/LightSkyDefense/Scripts/Enemies/Abilities/FindTarget.cs:
```diff
@@ -0,0 +1,53 @@
+
+    private void FixedUpdate()
+    {
+        var TargetsInRange = Physics.OverlapSphere(transform.position, Radius, LayerBitMask);
```
Divide to multiple lines?

Maxboef commented on src/Assets/LightSkyDefense/Scripts/Enemies/Abilities/EnemyAoEHeal.cs:
```diff
@@ -0,0 +1,48 @@
+
+public class EnemyAoEHeal : MonoBehaviour
+{
+    public float HealAmount = 0.1f;
```
Tooltips?

Commit:
Add tooltips #182

Commit:
Update comments #182

Commit:
Add clarify if statement in flocking #182
